src/hardware/hardware.cpp

```diff
@@ -507,6 +522,7 @@ void CAPTURE_AddImage(Bitu width, Bitu height, Bitu bpp, Bitu pitch, Bitu flags,
 		case 8:format = ZMBV_FORMAT_8BPP;break;
 		case 15:format = ZMBV_FORMAT_15BPP;break;
 		case 16:format = ZMBV_FORMAT_16BPP;break;
+		case 24:format = ZMBV_FORMAT_32BPP;break;	// 24-bit will be converted to 32-bit for compatibility
```

Comment:
why exactly isn't this `ZMBV_FORMAT_24BPP`?

Reply:
The ZMBV library doesn't handle 24bit pixels (currently), so the S3 patch authors author wrote an integer cast operator so ZMBV can access the arrays thinking it's dealing with a 32bi format.

We could add 24-bit support. ZMBV is known as the "DosBox Capture Codec" (and supported by VLC and others) - but long term, it might make sense to drop ZMBV entirely and offload that to desktop/application recording software (and benefit from hardware acceleration/improved compression codecs/etc.)

Reply:
Added comment in the code:
```c++
// ZMBV is "the DOSBox capture format" supported by external
// tools such as VLC, MPV, and ffmpeg. Because DOSBox originally
// didn't have 24-bit color, the format itself doesn't support
// it. I this case we tell ZMBV the data is 32-bit and let the
// rgb24's int() cast operator up-convert.
case 24: format = ZMBV_FORMAT_32BPP; break;
```